release/src/router/rc/wanduck.c

```diff
@@ -2340,8 +2340,8 @@ void handle_dns_req(int sfd, unsigned char *request, int maxlen, struct sockaddr
 #endif
 	unsigned char reply_content[MAXLINE], *ptr, *end;
 	dns_header *d_req, *d_reply;
-	dns_queries queries;
-	dns_answer answer;
+	dns_queries queries = {0};
+	dns_answer answer = {0};
```

Comment:
initing with {0} is not necessary, there's memset(&answer 0, sizeof(answer)) down below

Reply:
I would've removed the memset(), but reading the comment I assumed you planned for implementing handling of multiple queries so I left it there.